packages/react/src/components/Slider/Slider.test.tsx

```diff
@@ -444,6 +444,34 @@ describe('Slider', () => {
     jest.runOnlyPendingTimers();
   });
 
+  it('onChanged returns the correct value', () => {
+    jest.useFakeTimers();
```

Comment:
Some small things which can be done in a follow-up PR:

- Add this in the `afterEach` to ensure it goes back to using real timers afterwards
```
    if ((setTimeout as any).mock) {
      jest.useRealTimers();
    }
```
- Add a test that onChanged gets the correct value after dragging with the mouse (could maybe combine with the existing sliding test)
- Add a test that onChanged gets the correct value if the user clicks and releases the thumb without moving. (It's kind of debatable whether onChanged should be called at all in that case, but historically it has been, so it's good to have a test to document that behavior and ensure it continues working.)